files/image_config/rsyslog/rsyslog.d/00-sonic.conf

```diff
@@ -13,12 +13,12 @@ if re_match($programname, "bgp[0-9]*#bgpd") then {
 ## Teamd rules
 
 if $programname contains "teamd_" then {
-    /var/log/teamd.log
+    action(type="omfile" file="/var/log/teamd.log" flushOnTXEnd="off" asyncWriting="on" flushInterval="5" ioBufferSize="64k")
```

Comment:
>flushOnTXEnd [](http://example.com/codeflow?start=51&length=12)

Compare with previous config (you may know the default values), what changed?

Reply:
https://github.com/praveen-li/equilibrium/blob/main/rsyslog/rsyslog_config.md
Above page has details and testing. 

In summary, we are adding an action queue for each facility (say teamd), which has 64k buffer size for logs. Logs are written to SSD only 
- when buffer is full or,
- 5 secs are passed, or
- rsyslog\system shutdown.

This is done to improve SSD life, since SSD comes with fixed erase\write counts.